app/models/spree/payment_decorator.rb

```diff
@@ -1,5 +1,5 @@
 Spree::Payment.class_eval do
-  before_update :send_gift_card, :if => "state_changed? && state_was != 'completed' && state == 'completed'"
+  after_update :send_gift_card, :if => "state_changed? && state_was != 'completed' && state == 'completed'"
```

Comment:
@AnuragBhardwaj1  Please move the condition to a separate method. Plus do ensure we are sending mail considering ActiveJob